test/new-e2e/pkg/components/remotehost.go

```diff
@@ -130,6 +132,42 @@ func (h *RemoteHost) RemoveAll(path string) error {
 	return clients.RemoveAll(h.client, path)
 }
 
+func (h *RemoteHost) TunnelTCP(localport, remoteport uint16) (shutdown func() error, err error) {
+	l, err := net.Listen("tcp", fmt.Sprintf("127.0.0.1:%d", localport))
+	if err != nil {
+		return nil, err
+	}
+
+	go func() {
+		for {
+			lc, err := l.Accept()
+			if err != nil {
+				fmt.Printf("ERROR ACCEPTING: %v\n", err)
+				return
+			}
+			go func() {
+				fmt.Printf("Tunneling %d -> %d", localport, remoteport)
+				defer lc.Close()
+				rc, err := h.client.Dial("tcp", fmt.Sprintf("127.0.0.1:%d", remoteport))
+				if err != nil {
+					fmt.Printf("Failed to connect to remote: %v\n", err)
+					//	h.client.
+				}
+				defer rc.Close()
+				go func() {
+					fmt.Printf("Copying remote -> local.\n")
+					defer fmt.Printf("Done Copying remote -> local.\n")
+					io.Copy(rc, lc)
+				}()
+				fmt.Printf("Copying local -> remote.\n")
+				defer fmt.Printf("Done Copying local -> remote.\n")
+				io.Copy(lc, rc)
+			}()
+		}
+	}()
+	return l.Close, nil
+}
+
```

Comment:
This might be useful for the future, even if we don't want it right now.
It allows us to create a TCP tunnel into the host and communicate with things on the remote.

Initially, I was going to use this to run a tracer locally and send traces to the remote agent, before discussions determined that we'd need a remote tracer anyway.

I won't put it into this PR, but it's here now to see if it's something I should move into another PR or just delete.
WDYT?